app/common/auth/decorators.py

```
@@ -67,11 +69,35 @@ def platform_admin_role_required[**P](
     def wrapper(*args: P.args, **kwargs: P.kwargs) -> ResponseReturnValue:
         # This decorator is itself wrapped by `mhclg_login_required`, so we know that `current_user` exists and is
         # not an anonymous user (ie a user is definitely logged-in) and an MHCLG user if we get here.
-        user = get_current_user()
+        user = interfaces.user.get_current_user()
 
-        if not user.is_platform_admin:
+        if not AuthorisationHelper.is_platform_admin(user):
             abort(403)
 
         return func(*args, **kwargs)
 
     return mhclg_login_required(wrapper)
+
+
+def has_grant_role[**P](
+    role: RoleEnum,
+) -> Callable[[Callable[P, ResponseReturnValue]], Callable[P, ResponseReturnValue]]:
+    def decorator(func: Callable[P, ResponseReturnValue]) -> Callable[P, ResponseReturnValue]:
+        @functools.wraps(func)
+        def wrapped(*args: P.args, **kwargs: P.kwargs) -> ResponseReturnValue:
+            current_user = interfaces.user.get_current_user()
+
+            if AuthorisationHelper.is_platform_admin(user=current_user):
+                return func(*args, **kwargs)
+
+            if "grant_id" in kwargs:
```

If we add the `has_grant_role` decorator to an endpoint that defines the variable as anything other than `grant_id` (accidentally or maybe for some unexpected reason), this check would get skipped altogether.

That feels like an insecure default.

Should we raise an error if we don't find a `grant_id` in the kwargs? That will help ensure consistency and always be a more "fail closed" secure default.

---

Nice idea, yes lets do this

---

yeah since this explicitly check for grant role before that what my idea was different let me update

---

Fixed